packages/pytest-simcore/src/pytest_simcore/helpers/playwright.py

```diff
@@ -106,6 +112,94 @@ class SocketIOEvent:
 SOCKETIO_MESSAGE_PREFIX: Final[str] = "42"
 
 
+@dataclass
+class RestartableWebSocket:
+    page: Page
+    ws: WebSocket
+    _registered_events: list[tuple[str, typing.Callable | None]] = field(
```

Comment:
MINOR. why not a `collections.OrderedDict[str, Callable | None]`  instead of a `list`? This way you make sure they are e.g. not duplicated by design

Reply:
My tests are almost green :) in the next PR coming with additional improvement I can add it there